docs/sources/flow/tutorials/flow-by-example/processing-logs/index.md

```diff
@@ -0,0 +1,382 @@
+
+### Stage 3
+
+```river
+stage.json {
+    source = "log"
+
+    expressions = {
+        is_secret = "",
+        level     = "",
+        log_line  = "message",
+    }
+}
+```
+
+This stage is acting on the `log` value in the extracted map, which is a value that we extracted in the previous stage. This value also happens to be a JSON object, so we can extract values from it as well. This stage extracts three values from the `log` value, `is_secret`, `level`, and `log_line`, and puts them into the extracted map with keys `is_secret`, `level`, and `log_line`.
+
+If it were Python, it would be roughly equivalent to:
+
+```python
+extracted_map = {
+    "log": {
+        "is_secret": "true",
+        "level": "info",
+        "message": "This is a secret message!",
+    },
+    "ts": "2023-11-16T06:01:50Z",
+}
+
+source = extracted_map["log"]
+
+extracted_map["is_secret"] = source["is_secret"]
+extracted_map["level"]     = source["level"]
+extracted_map["log_line"]  = source["message"]
+```
+
+Extracted map _before_ performing this stage:
+
+```json
+{
+    "log": {
+        "is_secret": "true",
+        "level": "info",
+        "message": "This is a secret message!",
+    },
+    "ts": "2023-11-16T06:01:50Z",
+}
+```
+
+Extracted map _after_ performing this stage:
+
+```json
+{
+    "log": {
+        "is_secret": "true",
+        "level": "info",
+        "message": "This is a secret message!",
+    },
+    "ts": "2023-11-16T06:01:50Z",
+    "is_secret": "true",
+    "level": "info",
+    "log_line": "This is a secret message!",
+}
+```
+
+### Stage 4
+
+```river
+stage.drop {
+    source = "is_secret"
+    value  = "true"
+}
+```
+
+This stage is acting on the `is_secret` value in the extracted map, which is a value that we extracted in the previous stage. This stage drops the log line if the value of `is_secret` is `"true"` and does not modify the extracted map. There are many other ways to filter logs, but this is a simple example. Refer to the [loki.process#stage.drop](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/loki.process/#stagedrop-block) documentation for more information if you are curious.
+
+### Stage 5
+
+```river
+stage.labels {
+    values = {
+        level = "",
+    }
+}
+```
+
+This stage adds a label to the log using the same shorthand as above (so this is equivalent to using `values = { level = "level" }`). This stage adds a label with key `level` and the value of `level` in the extracted map to the log (`"info"` from our example log line). This stage does not modify the extracted map.
+
+### Stage 6
+
+```river
+stage.output {
+    source = "log_line"
+}
+```
+
+This stage uses the `log_line` value in the extracted map to set the actual log line that is forwarded to Loki. Rather than sending the entire JSON blob to Loki, we are only sending `original_log_line["log"]["message"]`, along with some labels that we attached.
```

Comment:
```suggestion
This stage uses the `log_line` value in the extracted map to set the actual log line that is forwarded to Loki.
Rather than sending the entire JSON blob to Loki, you are only sending `original_log_line["log"]["message"]`, along with some labels that you attached.
```